File: Data Structures & Algorithms/maximum-product-subarray/submission-3.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int curr_max = nums[0];
        int curr_min = nums[0];
        int ans = nums[0];
        int temp = 0;
        for (int i = 1; i < nums.size(); i++) {
            if (nums[i] < 0) {
                temp = curr_max;
                curr_max = curr_min;
                curr_min = temp;
            }
            curr_max = max(nums[i], nums[i] * curr_max);
            curr_min = min(nums[i], nums[i] * curr_min);
            ans = max(curr_max, ans);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-3.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        min_val = min(nums)
        max_val = max(nums)
        nums_bin = [0] * (max_val - min_val + 1)

        for num in nums:
            nums_bin[num - min_val] += 1
        
        for i in range(len(nums_bin) - 1, -1, -1):
            k -= nums_bin[i]
            if k <= 0:
                return i + min_val